core/src/main/java/org/infinispan/persistence/file/SingleFileStore.java

```diff
@@ -505,38 +583,31 @@ public void process(KeyFilter<? super K> filter, final CacheLoaderTask<K, V> tas
             if (filter.accept(e.getKey()))
                keysToLoad.add(new KeyValuePair<>(e.getKey(), e.getValue()));
          }
-         Collections.sort(keysToLoad, new Comparator<KeyValuePair<K, FileEntry>>() {
-            @Override
-            public int compare(KeyValuePair<K, FileEntry> o1, KeyValuePair<K, FileEntry> o2) {
-               long offset1 = o1.getValue().offset;
-               long offset2 = o2.getValue().offset;
-               return offset1 < offset2 ? -1 : offset1 == offset2 ? 0 : 1;
-            }
+         keysToLoad.sort((o1, o2) -> {
```

Comment:
@wburns The entries map is not a `ConcurrentHashMap`, it's a `[Linked]HashMap`, so iterating through the entry set (or any read for that matter) needs to be synchronized.